tinygrad/jit.py

```diff
@@ -55,14 +56,24 @@ def __call__(self, *args, **kwargs) -> Any:
         for i,a in enumerate(cache[1]):
           if a in [v[0] for v in input_rawbuffers.values()]:
             self.input_replace[(j_,i)] = [(k, v[1], v[0].dtype) for k,v in input_rawbuffers.items() if v[0] == a][0]
+            self.updatable_entries[j_].append(i)
+        for i in range(len(cache[2])): self.updatable_entries[j_].append(len(cache[1])+i)
         #if prg.local_size is None: prg.local_size = prg.optimize_local_size(args, preserve_output=True)  # the JIT can optimize local
       assert set([x[0] for x in self.input_replace.values()]) == set(input_rawbuffers.keys()), "some input tensors not found"
+
+      # init batch_executor
+      try: self.batch_executor = cast(Compiled, Device[Device.DEFAULT]).batch_exec(self.jit_cache)
```

Comment:
Hmm...should this be the DEFAULT Device? Also, shouldn't BasicBatchExecutor just be the default on compiled?

Reply:
BasicBatchExecutor is the default now. Possible to take batch_exec from jit_cache's prg itself, is it better? Anyways HipGraph and others check if they can serve given jit_cache overwise it switches to BasicBatchExecutor.